lib/sles4sap_publiccloud.pm

```diff
@@ -182,6 +184,70 @@ sub sles4sap_cleanup {
 
 }
 
+=head2 get_job_state
+    Get whether a specified job is still running or not
+=cut
+
+sub get_job_state {
+    my ($job_id) = @_;
+
+    my $url = "https://openqa.suse.de/api/v1/jobs/$job_id";
+
+    my $json_data = script_output("curl -s '$url'");
+
+    my $job_data = eval { decode_json($json_data) };
+    if ($@) {
+        warn "Failed to decode JSON data for job $job_id: $@";
+        return 0;    # Assume job is still running if we can't get its state
+    }
+
+    my $job_state = $job_data->{job}->{state} // '';
+
+    if ($job_state eq 'running') {
+        return 0;    # Job is still running
+    }
+    else {
+        return 1;    # Job has stopped
+    }
+}
+
+
+=head2 get_peerings_azure
+    Get active peerings for Azure jobs
+=cut
+
+sub get_peerings_azure {
+    my ($self) = @_;
+    my $cmd = 'az network vnet peering list -g dsdm0we --vnet-name vnet-dsdm0we --query "[].name"';
+    my $output_str = script_output($cmd);
+    my @output = split(/\n/, $output_str);
+    my %result;
+
+    foreach my $line (@output) {
+        my @matches = $line =~ /(\d{6,})/g;
+        $result{$line} = $matches[-1] if @matches;
+    }
+    return %result;
+}
+
+=head2 clean_peerings_azure
+    Delete leftover peerings for Azure jobs that finished without cleaning up
+=cut
+
+sub clean_peerings_azure {
```

Comment:
Maybe a function in this layer, named `clean_all_peerings_azure` can stay in this layer, to read `IBSM_RG` and to call all the `az` functions from `lib/qesapdeployment.pm`